### ISSUE TITLE
`sudo: false` allows sudo commands

### ISSUE THREAD

**Issue by GiedriusM** [travis-ci/travis-ci#8877]
Setting `sudo: false` in .travis.yml file still allows running sudo commands. Which indicates either that VM is used despite sudo setting, or that the docker is running with root access.
Testing repo: https://github.com/GiedriusM/travis-test
.travis.yml: https://github.com/GiedriusM/travis-test/blob/447a7f305a3f9b4bc52274be9803fc7f202072e4/.travis.yml#L1-L12
Travis build: https://travis-ci.org/GiedriusM/travis-test/builds/314082723

Some of the commands were used based on https://stackoverflow.com/questions/20010199/determining-if-a-process-runs-inside-lxc-docker and their results suggests, that `sudo: false` is running on a VM and not a docker.

Original testing started here: https://github.com/travis-ci/travis-ci/issues/8361#issuecomment-350090030

**Comment by BanzaiMan** [comment]
Your build explicitly calls `sudo`, so we route your jobs to the infrastructure that can accommodate it.

**Comment by GiedriusM** [comment]
Does `sudo` option really have a purpose then if it's overwritten by autodetection? [Documentation](https://docs.travis-ci.com/user/reference/overview/) makes it look like it should switch between container or VM. I guess this can be closed if it's not a bug, although I would prefer an error instead of under-the-hood reconfiguration.

**Comment by edmorley** [comment]
It looks like the check here should make sure the specified value is empty before setting the default?
https://github.com/travis-ci/travis-api/blob/180c4ef92bce286a35df7f46ee15f50b54ddafeb/lib/travis/model/job/queue.rb#L97

**Comment by GiedriusM** [comment]
> It looks like the check here should make sure the specified value is empty before setting the default?

Yeah. Maybe it does, because build [config](https://travis-ci.org/GiedriusM/travis-test/jobs/314082724/config ) says it's using the `sudo: false` setting:
```json
{
  "dist": "trusty",
  "sudo": false,
  "before_script": "sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6'",
  "script": [
    "sudo sh -c 'lsmod'",
    "sudo sh -c 'rmmod dm_crypt'",
    "sudo sh -c 'lsmod'",
    "sudo sh -c 'cat /proc/1/cgroup'",
    "sudo sh -c 'cat /proc/1/environ'",
    "sudo sh -c 'ls -la /'",
    "sudo sh -c 'ls -la /bin'",
    "sudo sh -c '/bin/running-in-container; echo $?'"
  ],
  "language": "ruby",
  "group": "stable",
  "os": "linux"
}
```
But in actuality it's not.
